### PR TITLE
Rename PropertyIndex to EntityIndex.

### PR DESCRIPTION
**Pull request by arthurp**
With: https://github.com/KatanaGraph/katana-enterprise/pull/3026

This is a mechanical rename PR. There should be no true code changes.

### CODE REVIEW

**Review by chakpongchung — APPROVED** [review]
Though it is not a reason not to merge this PR first, would you mind holding off this PR until maybe a few more days to make things a little easier on my side?

Otherwise, looks good to me

**Review by chakpongchung — APPROVED** [review]

File: libgraph/include/katana/PropertyGraph.h
Comment (chakpongchung) [inline libgraph/include/katana/PropertyGraph.h:898]: @arthurp We need this function exposed to Python users. I can do that or maybe you are doing that already? (Not sure whether this is the reason you are renaming things to make things easier)

Currently, python users seem not able to check the existence of
the node index. Without this, we cannot correctly test double-create
or double-delete cases properly for indexing in pytest. Turns out one specific order
of running index-related tests worked for a while without proper graph
state management. @roshandathathri

Comment (arthurp) [reply libgraph/include/katana/PropertyGraph.h:898]: The rename is because "PropertyIndex" now refers to an index into a property array, since that will not be the same as the node or edge ID (soon).

This feature is not exposed in Python yet. I'm not even sure I have a request for it. Please create a JIRA task and assign it to me if you need it. I'm working on PropertyGraph wrappers now, so I might be able to make it happen as part of that work.

**Comment by arthurp** [comment]
> Though it is not a reason not to merge this PR first, would you mind holding off this PR until maybe a few more days to make things a little easier on my side?
> 
> Otherwise, looks good to me

I would assume that one of us will need to do a search and replace to fix the issue, if you are using these methods and types. I can wait, but it doesn't seem like it would be any easier for me to search and replace than for you to do it.

File: libgraph/include/katana/PropertyGraph.h
Comment (chakpongchung) [reply libgraph/include/katana/PropertyGraph.h:898]: @arthurp 
Just created a JIRA ticket. The tests of interest here are here:
https://github.com/KatanaGraph/katana-enterprise/blob/master/python/test/remote_graph_test.py#L283

Talking to @arekay  to confirm whether this is truly wanted. At least it is a bug to me when refactoring the RunQuery/LoadGraph within QueryClient.

@roshandathathri

**Comment by chakpongchung** [comment]
> > Though it is not a reason not to merge this PR first, would you mind holding off this PR until maybe a few more days to make things a little easier on my side?
> > Otherwise, looks good to me
> 
> I would assume that one of us will need to do a search and replace to fix the issue, if you are using these methods and types. I can wait, but it doesn't seem like it would be any easier for me to search and replace than for you to do it.

Yes, I am using some of the methods here. I will appreciate if you can wait maybe until Jan 26th unless this renaming is blocking you. After that renaming should have no impact on me.

**Comment by arthurp** [comment]
> > > Though it is not a reason not to merge this PR first, would you mind holding off this PR until maybe a few more days to make things a little easier on my side?
> > > Otherwise, looks good to me
> > 
> > 
> > I would assume that one of us will need to do a search and replace to fix the issue, if you are using these methods and types. I can wait, but it doesn't seem like it would be any easier for me to search and replace than for you to do it.
> 
> Yes, I am using some of the methods here. I will appreciate if you can wait maybe until Jan 26th unless this renaming is blocking you. After that renaming should have no impact on me.

I have to admit it seems like you are asking me to predicate my work on yours just so you can avoid some find and replace. This seems a bit silly. That said, I am not immediately blocked on this, I can just remove some features from the work I'm doing until this merges. Ironically those features are wrapping this API for python which is something you want. ;-)

**Comment by roshandathathri** [comment]
> I have to admit it seems like you are asking me to predicate my work on yours just so you can avoid some find and replace. This seems a bit silly.

Agree with Arthur. Merge conflicts can occur but we shouldn't wait to merge PRs unless it's critical. 

> Currently, python users seem not able to check the existence of the node index. 

As discussed in the query meeting today, I'm not sure whether this is required/useful. We should hold off on making changes to the Python API without a wider discussion. 

Arthur, in summary, please feel free to merge this PR.

File: libgraph/CMakeLists.txt
Comment (roshandathathri) [inline libgraph/CMakeLists.txt]: nit: clang-tidy didn't catch this spacing issue?

**Review by roshandathathri — APPROVED** [review]
LGTM. Thanks!

nit: Please modify the commit message when you merge to include the search-replace rules used so that it is easy for others to do it if they encounter conflicts. Something like:
PropertyIndex -> EntityIndex
HasNodePropertyIndex -> HasNodeIndex

File: libgraph/CMakeLists.txt
Comment (arthurp) [reply libgraph/CMakeLists.txt]: Ooops. There is no format checking for cmake code. :-) I'll fix it.

Can I get a review on https://github.com/KatanaGraph/katana-enterprise/pull/3026 as well. It's just updating the users in enterprise.

**Comment by arthurp** [comment]
> Please modify the commit message when you merge to include the search-replace rules used so that it is easy for others to do it if they encounter conflicts. Something like:
> PropertyIndex -> EntityIndex
> HasNodePropertyIndex -> HasNodeIndex

This is a good idea. I'll do what I can. I actually used a refactor in my IDE, but I think a search-and-replace would actually have done it too and I should be able to reconstruct the rules involved. The only complexity is that "PropertyIndex -> EntityIndex" is not valid. It meant two different things before this PR.

**Comment by roshandathathri** [comment]
> This is a good idea. I'll do what I can. I actually used a refactor in my IDE, but I think a search-and-replace would actually have done it too and I should be able to reconstruct the rules involved. The only complexity is that "PropertyIndex -> EntityIndex" is not valid. It meant two different things before this PR.

Thanks!

**Comment by chakpongchung** [comment]
> > Please modify the commit message when you merge to include the search-replace rules used so that it is easy for others to do it if they encounter conflicts. Something like:
> > PropertyIndex -> EntityIndex
> > HasNodePropertyIndex -> HasNodeIndex
> 
> This is a good idea. I'll do what I can. I actually used a refactor in my IDE, but I think a search-and-replace would actually have done it too and I should be able to reconstruct the rules involved. The only complexity is that "PropertyIndex -> EntityIndex" is not valid. It meant two different things before this PR.

Hi Arthur! This is exactly the reason why I wanted to see this renaming after, if possible, because you and I use the same IDE. Renaming, with the help of the IDE,  normally will not need to run ctest or integration locally again to verify things. As we all know, these two are painfully slow. 


As I mentioned here: https://github.com/KatanaGraph/katana/pull/827#pullrequestreview-860262194, I hope there is no hard feelings on this delay. 


@arthurp @roshandathathri

**Review by chakpongchung — APPROVED** [review]

**Comment by arthurp** [comment]
> Renaming, with the help of the IDE, normally will not need to run ctest or integration locally again to verify things. As we all know, these two are painfully slow.

I would think you would either be running this testing anyway (and you can put the rename in with it), or you don't need to run them since you only made the rename change in your most recent commit to your PR. So I think you can short cut this.

Also this is an issue where the compiler will catch issues fast, so you really only need to build to check that you got the rename correct.

**Pull request merged**
